src/commands/add.rs

```diff
@@ -68,13 +68,10 @@ fn merge_requirements(old: &Requirement, new: &Requirement) -> Option<Requiremen
     }
     merged.extras.extend(new.extras.iter().cloned());
     merged.extras.dedup();
-    if let Some(new_marker) = new.marker.as_ref() {
-        if old.marker.is_some() && old.marker.as_ref() != Some(new_marker) {
-            return None;
-        } else {
-            merged.marker = Some(new_marker.clone());
-        }
+    if old.marker != new.marker {
```

Comment:
this isn't exactly the same thing. the old code is saying if there exists a new marker, use that if there is no old marker or if its the same. the new code here wouldn't produce the same result. for example:

```rust
old.marker = None
new.marker = Some(marker)
// old code
merged.marker = Some(marker)
// new code
return None
```

Reply:
The `marker` field isn't an `Option<T>` anymore 🤐